tests/core/helpers.py

```diff
@@ -130,15 +135,19 @@ class TestModelFuture(ModelTrainerBase.ModelFutureBase):
 
         def __init__(self, parent, trained_model, save_path, save_with_id):
             super().__init__(
-                trainer_name=parent.instance_name,
-                training_id=str(uuid.uuid4()),
-                save_path=save_path,
+                future_name=parent.instance_name,
+                future_id=str(uuid.uuid4()),
```

Comment:
I think this should be reverted since we don't share future types. You could yank this class into a generic `JobFuture` and then subclass with the naming override?

Reply:
Hmmm, TestModelFuture and TestJobFuture share a couple of implementations but they have enough different to where each class would need a good amount of overrides (like each having to still implement get_info, result, and save)